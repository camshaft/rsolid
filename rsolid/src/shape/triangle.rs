use crate::*;

pub fn right(width: f64, height: f64) -> Object<2> {
    let angle = (height).atan2(width).to_degrees();
    let s = square([width, height]);
    let mask_v = width.max(height) * 2.0;
    let mask = square(mask_v) >> rotate_z(angle);
    let tri = s - mask;
    // center it and orient it
    tri >> back(height / 2.0) >> left(width / 2.0) >> mirror([1, 0, 0])
}

#[test]
fn right_test() {
    assert_2d_snapshot!(right(100.0, 200.0));
}

pub fn equilateral(length: f64) -> Object<2> {
    circle(3.0f64.sqrt() / 3.0 * length).fragment_resolution(3) >> rotate_z(-30)
}

#[test]
fn equilateral_test() {
    assert_2d_snapshot!(equilateral(80.0));
}
