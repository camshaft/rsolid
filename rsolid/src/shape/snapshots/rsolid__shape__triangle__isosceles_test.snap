---
source: rsolid/src/shape/triangle.rs
expression: "isosceles(100.0, 200.0)"
---
function _v0() = 3;
function _v1() = 57.735026918962575;
module _v2() { circle($fn=_v0(), r=_v1()); }
module _v3 (a=0) { rotate([0, 0, a]) children(); }
function _v4() = -30;
module _v5() { _v3(a=_v4()) children(); }
module _v6() { _v5() _v2(); }
module _v7 (y=undef) { translate([0, y, 0]) children(); }
function _v8() = -56.69872981077807;
module _v9() { _v7(y=_v8()) children(); }
module _v10() { _v9() _v6(); }

_v10();
